source/server/connection_handler_impl.h

```diff
@@ -252,9 +252,6 @@ class ConnectionHandlerImpl : public Network::ConnectionHandler,
                                             listener_.parent_.dispatcher_.timeSource(),
                                             StreamInfo::FilterState::LifeSpan::Connection)) {
       listener_.stats_.downstream_pre_cx_active_.inc();
-      stream_info_->setDownstreamLocalAddress(socket_->localAddress());
```

Comment:
What happened to these ones? Are they never needed?

Reply:
These are pure duplicates. They are set previously here right before the constructor runs: https://github.com/envoyproxy/envoy/pull/14432/files#diff-fe538327af200510b78cadc759f2c3af989c822a370d3f8a005b92bc59568aa5R465

(Please check me on this)

Reply:
I *think* (but I'm not sure) that the other calls happen AFTER the listener filters have run, and these ones ran before. I'm unsure if listener filters ever read the addresses out of the streaminfo though.

Reply:
Also, this area of confusion is the rationale for trying to not copy the values, but always get them from the connection, because it's so hard to reason about when these get set on various objects, and reflected between them.

Reply:
OK right, thanks for catching this. I was confusing ActiveTcpSocket with ActiveTcpConnection. It's also concerning that no test failed when I did this (I think this would mostly effect logging but I'm not sure). Let me look at this more and see if at least I can add a test that would catch this but maybe see what a shared_ptr solution would look like.

/wait

Reply:
I think this only affects what listener filters can view. But I just checked, and `ListenerFilterCallbacks` doesn't give access to `StreamInfo`, maybe it is ok to omit this. Can you construct any test that would fail with this call missing?

Reply:
It's not just what listener filters view, it effects logging if a listener filter closes a socket.

Yes I will definitely have a test, but I will also try to implement the shared_ptr thing and see how it goes.